src/Controller/DataMembershipController.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace App\Controller;
+
+use App\DataForm\DataMembership;
+use App\Form\DataMembershipType;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException;
+use Symfony\Component\HttpFoundation\File\Exception\FileException;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+use Symfony\Component\Routing\Annotation\Route;
+
+class DataMembershipController extends AbstractController
+{
+    /**
+     * @Route("/new_membership", name="new_membership")
+     */
+    public function new(
+        Request $request
+    ): Response {
+        $membership = new DataMembership();
+        $form = $this->createForm(DataMembershipType::class, $membership);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            /** @var UploadedFile $pdfFile */
+            $pdfFile = $form->get('membershipFile')->getData();
```

Comment:
Your field is mapped so you'd better access it via $membership->getMembershipFile() rather than dig in your form manualy

Reply:
Much more elegant indeed,
Thank you!